### ISSUE TITLE
Unable to run Kubernetes `v1.12.10`

### ISSUE THREAD

**Issue by eggsbenjamin** [kubernetes-sigs/kind#2270]
**What happened**:

```
$ kind create cluster --config=kind/config.yaml --name=integrated-test-local
```

```
# kind/config.yaml

kind: Cluster
apiVersion: kind.x-k8s.io/v1alpha4
containerdConfigPatches:
- |-
  [plugins."io.containerd.grpc.v1.cri".registry.mirrors."registry.registry:5000"]
    endpoint = ["http://registry.registry:5000"]
nodes:
- role: control-plane
  image: kindest/node:v1.12.10
  kubeadmConfigPatches:
  - |
    kind: InitConfiguration
    nodeRegistration:
      kubeletExtraArgs:
        node-labels: "ingress-ready=true"
  extraPortMappings:
  - containerPort: 80
    hostPort: 80
    protocol: TCP
  - containerPort: 443
    hostPort: 443
    protocol: TCP
```

The `coredns` pods crashloop with error

```
2021/05/23 17:05:27 [INFO] plugin/reload: Running configuration MD5 = f65c4821c8a9b7b5eb30fa4fbc167769
2021/05/23 17:05:34 [FATAL] plugin/loop: Seen "HINFO IN 1179812668599535537.130965306082261719." more than twice, loop detected
```

**What you expected to happen**:

The `coredns` pods should not crashloop _or_ I should be able to run `kindest/node:v1.12.10` with a specific version of `kind`.

**How to reproduce it (as minimally and precisely as possible)**:

Sping up the cluster using the above command and config.

**Anything else we need to know?**:

A [similar issue](https://github.com/kubernetes-sigs/kind/issues/1845) exists in which it's suggested that this breaking change was introduced in `v0.8.x` but it seems that the error exists in versions prior to this.

**Environment:**

- kind version: (use `kind version`):
  - `v0.7.0`
  - `v0.8.1`
  - `v0.9.0`
- Kubernetes version: (use `kubectl version`): `v1.12.10`
- Docker version: (use `docker info`): `20.10.5`
- OS (e.g. from `/etc/os-release`): macOS Catalina 10.15.7

**Comment by BenTheElder** [comment]
It's not supported and will not be. In previous release notes we've mentioned this.

**Comment by BenTheElder** [comment]
We also don't patch or support older releases of kind at this stage, we just do not have the bandwidth fornthis. Kind is alpha and not branched. 0.11 is current.

The Kubernetes project itself only supports 1.19+ now. As a sub project we've opted to best effort support older releases. Currently that is 1.14+ with CI covering 1.17+. Even running CI that old has been contentious.

**Comment by BenTheElder** [comment]
Additionally it is not supported to run arbitrary images with arbitrary kind versions and never has been. This is covered in the release notes as well. Our prepublished images are documented, other images may have or be lacking breaking changes necessary to continue improving the project.


If you need to use a 1.12 image you'll have to use the kind release for which it was published (it will be listed in the release notes. We do not support 1.12 in current versions and it not working is expected.

In the future we will likely switch to a scheme where prepublished image tags include the kind version to make it extra clear.

**Comment by eggsbenjamin** [comment]
@BenTheElder 

> If you need to use a 1.12 image you'll have to use the kind release for which it was published

I'll take a look at the release not of the older versions of `kind`. Thanks.

> In the future we will likely switch to a scheme where prepublished image tags include the kind version to make it extra clear.

This would make things clearer. Kind is a great tool and for people responsible for clusters running legacy k8s versions having the ability to emulate this locally is very useful. It would be a shame for this tonot to be possible.

Cheers!

**Comment by BenTheElder** [comment]
> Kind is a great tool and for people responsible for clusters running legacy k8s versions having the ability to emulate this locally is very useful. It would be a shame for this to not be possible.

I agree, but kubeadm in particular has matured quite a bit since Kubernetes v1.12, especially by 1.14+ and we now depend on those improvements.

For context when KIND was first started roughly 3 years ago it was intended only to be used for testing Kubernetes itself, meeting our needs that other tools were not meeting. At the time that meant Kubernetes v1.11.0+, as in v1.11 kubeadm switched to supporting a config file and started moving away from the limited flags.

Since then we've only dropped support for 1.11, 1.12, and 1.13 as they've become problematic to continue working with the other dependencies. Averaging one version lost a year while 4 have been released each year. Eventually that will probably be unsustainable.

Kubernetes is currently working on 1.22, which kind is used to test in development and must support.

In general the Kubernetes project as a whole has only agreed to 3 versions, though those versions will be published more slowly now. Right now that would be 1.21, 1.20, 1.19. This also means we cannot patch any bugs that are in Kubernetes itself for older versions, which eventually becomes a problem.

https://minikube.sigs.k8s.io/docs/handbook/config/#selecting-a-kubernetes-version

https://kubernetes.io/releases/version-skew-policy/
